internal/service/eks/addon_version_data_source.go

```diff
@@ -0,0 +1,59 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceAddonVersion() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceAddonVersionRead,
+		Schema: map[string]*schema.Schema{
+			"addon_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"kubernetes_version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"latest": {
```

Comment:
Instead of `latest`, Terraform data sources use `most_recent` and sort by date traditionally. 

If the `DescribeAddonVersions` guarnatees that the first item in the addonVersions list is the latest, you could use that, but you should verify with the service team. I do worry about an addon supporting more than one major version at a time and incrementing the build version on the same day, how would `latest` or `most_recent` determine the proper one to return? [See this example](https://docs.aws.amazon.com/eks/latest/APIReference/API_DescribeAddonVersions.html#API_DescribeAddonVersions_Examples).

I would suggest sorting by Semantic Version, but you still run into the problem above. Feedback welcome.

Reference:
https://github.com/hashicorp/terraform-provider-aws/blob/a65002b873e17df5d0f19a6d16ff53d0c91a5aab/internal/service/ec2/ebs_snapshot_data_source.go#L43

Reply:
I had a look at the usage of `latest` vs `most_recent` in previous datasources and `most_recent` is indeed much more common so I'll change the code to use `most_recent`.

`most_recent=true` will always return the first item in the list of supported versions as they are always returned lexicographically sorted. Within the list of supported versions of an addon for a kubernetes version, only one of them is set as the default version for that kubernetes version (default=true). Therefore `most_recent=false` checks the entire list of supported versions and returns the one with `default` set to true.